File: BOJ/Study/Datastructure1-Reference/ROT13.py
# ROT13

import sys
s = sys.stdin.readline().rstrip()
for x in s:
    if ord(x) < 65:
        print(x, end="")
        continue
    if 64 < ord(x) < 91:
        tmp = ord(x) - 65
        tmp += 13
        tmp %= 26
        print(chr(tmp + 65), end="")
    else:
        tmp = ord(x) - 97
        tmp += 13
        tmp %= 26
        print(chr(tmp + 97), end="")

File: BOJ/Study-CPP/Datastructure1/ROT13.cpp
// ROT13

#include <iostream>
#include <string>
using namespace std;


int main() {
    string s;
    getline(cin, s);
    char tmp;
    for (int i = 0; i < s.length(); i++) {
        if (64 < s[i] && s[i] < 91) {
            tmp = ((int)s[i]%65 + 13)%26 + 65;
            cout << tmp;
        } else if(96 < s[i] && s[i] < 123) {
            tmp = ((int)s[i]%97 + 13)%26 + 97;
            cout << tmp;
        } else {
            cout << s[i];
        }
    }
    
    return 0;
}